core/src/main/scala/org/apache/spark/sql/SnappyBaseParser.scala

```diff
@@ -298,7 +300,8 @@ abstract class SnappyBaseParser(session: SparkSession) extends Parser {
   }
 
   protected final def columnCharType: Rule1[DataType] = rule {
-    VARCHAR ~ '(' ~ ws ~ digits ~ ')' ~ ws ~> ((d: String) => VarcharType(d.toInt)) |
+    VARCHAR ~ ('(' ~ ws ~ digits ~ ')').? ~ ws ~> ((d: Any) => VarcharType(
```

Comment:
@ahshahid there is another requirement to support just Varchar as the column specification. i.e. allow `create table (a int, b varchar, c varchar(10)) using column`. Seems like we could modify this rule to support this